### ISSUE TITLE
wrong 'variable is defined but never used' warning with JSX

### ISSUE THREAD

**Issue by mcharytoniuk** [eslint/eslint#2298]
After updating to eslint 0.19 it gives untrue warning about unused variable. The following code:

```JavaScript
"use strict";

import HelloWorld from "example.com/HelloWorld";
import React from "react";

React.render(<HelloWorld />, document.body);
```

Produces error:
```bash
10:7  error  HelloWorld is defined but never used  no-unused-vars 
```

My broken build:
https://travis-ci.org/goreutils/gore-gulp/jobs/58263707

My .eslintrc:
https://github.com/goreutils/gore-gulp/blob/8cafc0880a7334a128bebc791cfb6eec11f37deb/eslint/.eslintrc

**Comment by lo1tuma** [comment]
This is the correct behavior. ESLint supports only the JSX syntax not the semantic of react. There is a plugin which implements the semantic of react: [eslint-plugin-react](https://github.com/yannickcr/eslint-plugin-react).

**Comment by mcharytoniuk** [comment]
Thanks.

**Comment by nzakas** [comment]
Yeah, is likely that eslint-plugin-react needs updating. Please file an issue there.
